src/libraries/System.Private.CoreLib/src/System/IO/Enumeration/FileSystemEntry.Unix.cs

```diff
@@ -95,7 +99,7 @@ public ReadOnlySpan<char> FileName
             {
                 if (_directoryEntry.NameLength != 0 && _fileName.Length == 0)
                 {
-                    Span<char> buffer = MemoryMarshal.CreateSpan(ref _fileNameBuffer._buffer[0], Interop.Sys.DirectoryEntry.NameBufferSize);
+                    Span<char> buffer = MemoryMarshal.CreateSpan(ref _fileNameBuffer._buffer[0], DecodedNameBufferLength);
                     _fileName = _directoryEntry.GetName(buffer);
```

Comment:
```suggestion
                    _fileName = _directoryEntry.GetName(_buffer);
```
If you accept the suggestion above, this should be able to use default conversion from InlineArray to Span

Reply:
This failed for me locally:
> error CS9084: Struct member returns 'this' or other instance members by reference

I pushed anyways to share the CI error, it does compile fine in [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGADAAQY4B0ASgK4B2ALgJYC2MlAwhKwA7MAbGFADKwgG7MwMAM4BuQoQwBmUgxkBDAGYxSsLaRmMotMI1IAxQTAoBRJlACehAN6FSH0ryjNxGxrrUMBoAJgDy9AKOIrwa9AA8YAAWGlAAfKQA+lrWAHIa7AoEnl4+fgGWeQUwAEK0WjpQWTlC+ex1DcJFJd6+/roAIsywZtCO9saOWSHDMKNOE05F7p4rHgDaAJKRzPQwAIJQUBqOABRYAKxIAJQAumvkqkYmZpWt1R2ND27FJZ67ASg9A0AlIyVSWXBUGI3U8AF8Hg8VKQgqEIlEYnFEil0m8YG0YN8HiUAOYwRjEzw/P405pVdikAC801m83GDkclAA4uSCadsvTavVGtdYbTPBgAOx096FSkeBG/BWERWKAgAenV6xExlMjAAMicILRGKdDY5jYwANK7EKUMQAR1oMCYzBBd0IAOEwNB6m0umeetIQxGjDGi2cBGp/yY3pBKOC4Ui0ViCShGR5jD5mLTOIywGFwmu33I0pCMC0GloAkYclIisVQA).

Reply:
cc @dotnet/roslyn 

https://dev.azure.com/dnceng-public/public/_build/results?buildId=1067762&view=logs&j=826908f3-d824-5a25-6bc6-d6f6821a85ed&t=928c92d7-808f-5c75-12c9-d31e9c1b8f40